src/PowerBalancerAgent.cpp

```diff
@@ -541,26 +557,37 @@ namespace geopm
 
     void PowerBalancerAgent::MeasureRuntimeStep::sample_platform(PowerBalancerAgent::LeafRole &role) const
     {
-        for (int pkg_idx = 0; pkg_idx < role.m_num_domain; ++pkg_idx) {
-            auto &package = role.m_package[pkg_idx];
-            int epoch_count = role.m_platform_io.sample(role.m_count_pio_idx[pkg_idx]);
-            if (epoch_count > 1 &&
-                epoch_count != package.last_epoch_count &&
-                !package.is_step_complete) {
-                /// We wish to measure runtime that is a function of node
-                /// local optimizations only, and therefore uncorrelated
-                /// between compute nodes.
-                double total = role.m_sample_agg->sample_epoch_last(role.m_time_agg_idx[pkg_idx]);
-                double network = role.m_sample_agg->sample_epoch_last(role.m_network_agg_idx[pkg_idx]);
-                double ignore = role.m_sample_agg->sample_epoch_last(role.m_ignore_agg_idx[pkg_idx]);
-                double balanced_epoch_runtime =  total - network - ignore;
+        int period_count = role.m_sample_agg->get_period();
+        // Wait m_warm_periods before starting algorithm
+        if (period_count > role.m_warm_periods &&
+            period_count != role.m_last_period_count) {
+            for (int pkg_idx = 0; pkg_idx < role.m_num_domain; ++pkg_idx) {
+                double sum_net_ig = 0.0;
+                std::set<int> nested_core = role.m_platform_topo.domain_nested(GEOPM_DOMAIN_CORE, GEOPM_DOMAIN_PACKAGE, pkg_idx);
+                std::set<int> active_nested_core;
+                std::set_intersection(nested_core.begin(), nested_core.end(),
```

Comment:
Okay, I doubt this will be a performance bottle neck, as it is only done once per period and the set sizes will have only about 40 elements (number of cores on a socket)